cmd/influxd/run/server.go

```diff
@@ -392,6 +395,21 @@ func (s *Server) Open() error {
 	return nil
 }
 
+func (s *Server) executeShutdown(c <-chan struct{}) {
+	// TODO corylanou stop the server from coming back up
+	for {
+		select {
+		case <-c:
+			err := s.Close()
+			if err != nil {
+				log.Println(err)
+				os.Exit(1)
+			}
+			os.Exit(0)
+		}
+	}
+}
```

Comment:
This should also monitor against the `closing` channel so that the goroutine can exit on close. Not a big deal for `influxd` but for tests the goroutines will start to accumulate.